docs/tealscript-migration.md

```diff
@@ -0,0 +1,508 @@
+# TEALScript Migration Guide
+
+This document is up-to-date as of TEALScript v0.107.0 and Algorand TypeScript v1.0.0-beta71
+
+## Migration Table
+
+| TEALScript | Algorand TypeScript | Notes |
+| --- | --- | --- |
+| JS Object | JS Object | Algorand TypeScript does not yet support nested dynamic types in JavaScript objects. TEALScript allows one level of nesting. For nested dynamic types, see [Objects](#objects) |
+| JS Array | JS Array | Algorand TypeScript does not yet support nested dynamic types in JavaScript arrays. TEALScript allows one level ofnesting. For nested dynamic types, see [Arrays](#arrays) |
+| `EventLogger` | [`emit`](https://dev.algorand.co/reference/algorand-typescript/api-reference/index/functions/emit) ||
+| `BoxKey` | [`Box`](TOOD: link to box docs) | The crate method has new parameters as shown [here](TODO: link to box section) |
+| `Txn` | `Transaction` ||
+| `PayTxn` | `PaymentTxn` ||
+| `AppCallTxn` | `ApplicationCallTxn` ||
+| `KeyRegTxn` | `KeyRegistrationTxn` ||
+| `OnCompletion` | `OnCompleteAction` ||
+| Eliptic curve opcodes (i.e `ecAdd`) | Now under [`ElipticCurve`](TODO: link to EC docs) (i.e. `ElipticCurve.add`) ||
+| `GlobalStateKey` | `GlobalState` ||
+| `LocalStateKey` | `LocalState` ||
+| `GlobalStateMap` | Not yet supported ||
+| `LocalStateMap` | Not yet supported ||
+| `isOptedInToApp` and `isOptedInToAsset` | [`isOptedIn`](TODO: link to isOptedInDocs) ||
+| `this.txn` | [`Txn`](TOOD: link to Txn docs) ||
+| `verify...Txn` | `assertMatch` | `assertMatch` can be used on any txn type or any object |
+| `globals` | [`Global`](TODO: link to Global docs) ||
+| `StaticArray` | `FixedArray` | May not cover all cases. See the array section for more details |
+
+## Migrations
+
+### Objects
+
+If all the property types in your object are fixed-length then no migration is necessary outside of other migrations mentioned in this document. More complex objects might need to use a different class shown below, but
+it is strongly recommended to use fixed-length types whenever possible.
+
+TODO: describe different types of objects and when you need to use them over POJOs
+
+### Arrays
+
+If there are no nested dynamic types (i.e fixed length types or a single dynamic array of a fixed-length type) then no migration is
+necessary outside of other migrations mentioned in this document. More complex arrays might need to use a different class shown below, but
+it is strongly recommended to use fixed-length types whenever possible.
+
+TODO: describe different types of arrays and when you need to use them over native arrays
```

Comment:
Waiting until all the types have landed so I can get some hands on experience before writing this section